screens/media/CreatePost.tsx

```diff
@@ -1,40 +1,233 @@
+  const [isPickingVideo, setIsPickingVideo] = useState<boolean>(false);
+  const [imageContent, setImageContent] = useState<LazyStaticImage[]>([]);
+  const [isPickingImages, setIsPickingImages] = useState<boolean>(false);
+  const [isProcessingPost, setIsProcessingPost] = useState<boolean>(false);
+  const [previewPost, setPreviewPost] = useState<PostMock | undefined>(
+    undefined
+  );
+
+  const baseBgColor = useColorModeValue('lightMode.base', 'darkMode.base');
+
+  // Build the preview post whenever relevant state changes
+  useEffect(() => {
+    if (user === null) return;
+    setPreviewPost(
+      new PostMock(
+        user,
+        new Date(Date.now()),
+        textContent,
+        [],
+        [],
+        imageContent,
+        videoContent
+      )
+    );
+  }, [textContent, videoContent, imageContent]);
+
+  // Opens up a video picker and updates state
+  const pickVideo = async () => {
+    try {
+      setIsPickingVideo(true);
+
+      const result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.Videos,
+        allowsEditing: true,
+        aspect: [4, 3],
+      });
+
+      if (result.cancelled) return;
+
+      // Get the first frame as a thumbnail
+      const thumbnailResult = await VideoThumbnails.getThumbnailAsync(
+        result.uri
+      );
+
+      setVideoContent(
+        new LazyStaticVideo(
+          'mock/path',
+          'mock/path',
+          thumbnailResult.uri,
+          result.uri
+        )
+      );
+    } finally {
+      setIsPickingVideo(false);
+    }
+  };
+
+  // Opens up an image picker and update state
+  const pickImages = async () => {
+    try {
+      setIsPickingImages(true);
 
-  const verifyAndPost = async () => { };
+      const result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.Images,
+        allowsMultipleSelection: true,
+        aspect: [4, 3],
+      });
+
+      if (result.cancelled) return;
+
+      setImageContent(
+        result.selected.map(
+          (info) => new LazyStaticImage('mock/path', info.uri)
+        )
+      );
+    } finally {
+      setIsPickingImages(false);
+    }
+  };
+
+  // Builds the blobs and sends it to the backend
+  const post = async () => {
+    try {
+      setIsProcessingPost(true);
+
+      if (user === null) {
+        setIsProcessingPost(false);
+        return;
+      }
+
+      const videoBlob =
+        videoContent !== undefined
+          ? {
+            video: await (await fetch(videoContent.videoUrl!)).blob(),
+            thumbnail: await (await fetch(videoContent.thumbnailUrl!)).blob(),
+          }
+          : undefined;
+
+      const imageBlobs = await Promise.all(
+        imageContent.map(async (image) => (await fetch(image.imageUrl!)).blob())
+      );
+
+      createPost(user, textContent, undefined, imageBlobs, videoBlob);
+    } finally {
+      setIsProcessingPost(false);
+      navigation.navigate('Tabs', {
```

Comment:
Will we only allow posts to be made from the profile tab screen?

Reply:
No not at all, but for now the default behavior is to navigate to the user's profile as that is the only guaranteed location to view it currently. We'd definitely support more navigation modes in the future.